analysis/basestats/analyze.h

```diff
@@ -38,13 +46,81 @@ class Analysis {
             } else if(req.ts < counters["MinTs"]) {
                 counters["MinTs"] = req.ts;
             }
+            // plot CDF
+            sizes[req.size]++;
+            // stddev, scv
+            reqsizes.push_back(req.size);
+            // one hit wonders
+            std::vector<Request> v;
+            if (reqs.count(req.oid)) {
+                v = reqs[req.oid];
+            }
+            v.push_back(req);
+            reqs[req.oid] = v;
```

Comment:
These seem all like very expensive operations. If you want to keep track of vectors in a map, you can use that C++ maps use default constructors, so there will be always a valid vector at reqs[req.oid]. So you can directly push back on that.
But, I'm not following why we need these data structures in the first place.
1) <map> is much slower than <unordered_map>, O(log n) vs O(1)! We don't seem to use the property that maps are ordered here (req.oids are random strings for the most part), so use unordered_maps.
2) We are tracking too much information which slows us down. Here's a more efficient proposal: use an unordered_map, which maps oid to the tuple of (req_count, last_request_timestamp). With a simple additional counter, which we increment whenever we see a request with last_request_timestamp < 2days, we get effectively the number of large intervals. We use the map to look up the last_request_timestamp and immediately update it (so we don't need to keep much history / use much memory). If we then want to distinguish between cold cache hits and real one-hit-wonders, we can iterate over the unordered_map and subtract the cold cache hits of all objects that were requested more than  once (but that's not strictly necessary).